### ISSUE TITLE
[BUG] basePath is not set when passed as additional property during java client generation

### ISSUE THREAD

**Issue by coder82** [OpenAPITools/openapi-generator#3601]
I am using openapi generator gradle plugin 4.0.3.

In my yaml file, the basepath is meant to use relative values:
```
servers:
- url: /api/v1
  description: Main API endpoint
```
I am tryng to pass additional parameters like this:
[image1](https://i.imgur.com/2KLoaJ3.png)
In the kotlin case it works indeed my basepath overridden properly:
[image2](https://i.imgur.com/lbPpCWi.png)
But in Java this is not working (uses default library okhttp):
[image3](https://i.imgur.com/LsnL8Yr.png)
Basically it's using the default host.
I am using the 2 tasks in the image above.
This is the output i get if I debug what's being passed to the mustache engine during Java generation, I grep/filtered it with "basePath":
[image4](https://i.imgur.com/9CZ8l0S.png)

This is IMHO incorrect or inconsistent.
What do you reckon?

**Comment by jimschubert** [comment]
Looks like this isn't constrained to the Java client… `UrlPathUtils` has this method:

```
public static String getProtocolAndHost(URL url) {
    if (url == null) {
        return LOCAL_HOST;
    } else {
        String protocol = (url.getProtocol() == null) ? "http" : url.getProtocol();
        return protocol + "://" + url.getHost();
    }
}
```

This will turn all relative paths to the constant value `"http://localhost"`. I think we'll need to allow an option to set this server location for relative paths.  Ideally, if we read the spec doc via HTTP/HTTPS, we should also attempt to extract the base (`scheme://host:port`) from the spec doc's URL.

I think the _easiest_ fix would be to add another CLI option and set that value into the `GlobalSettings` properties bag and consume it via `UrlPathUtils`. I don't personally like that global state, but I think it would require quite a bit of refactoring to solve in a more robust way.

**Comment by jimschubert** [comment]
I've added the `help wanted` label, because I think it's a pretty good way for someone to get their feet wet in the project.

@OpenAPITools/generator-core-team any alternative options to solving the problem of configuring that default base to something other than `"http://localhost"`?

**Comment by wing328** [comment]
@coder82 I believe the basePath can be changed during runtime in the Java client. Would that meet your requirement?

**Comment by coder82** [comment]
Actually not, I developed an sdk downloader from the documentation page and I’d like to generate the client stubs with the basePath already preset.

**Comment by coder82** [comment]
Later realized that the customization options are very poor so what I ended up doing is to preprocess the whole yaml file filling dynamic variables in it. It was the only way to allow full control. At this point you could close thie as I solve it by other means.
